src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -139,14 +128,21 @@ def __init__(self, argv):
         self.argv = argv
 
     def get_devices(self):
-        all_devices = disk.get_devices()
         # remove devices with partitions
-        # XXX Should be optional when getting device info
-        for device, detail in all_devices.items():
-            if detail.get('partitions') != {}:
-                del all_devices[device]
-        devices = sorted(all_devices.items(), key=lambda x: (x[0], x[1]['size']))
-        return device_formatter(devices)
+        devices = [(device, details) for device, details in
+                       disk.get_devices().items() if details.get('partitions') == {}]
+        size_sort = lambda x: x[1]['size']
+        return self._device_formatter(sorted(devices, key=size_sort))
+
+    def _device_formatter(self, devices):
```

Comment:
why is there a need to move this into a method?

Reply:
It's only used in the Batch object. No need for a module global function.

Reply:
Its also quite tightly coupled to the output of ```get_devices()```

Reply:
I would say that if it doesn't need to be moved lets leave it? Lets focus on the bug only

Reply:
What's your argument for have the formatting function in the global scope?